services/omnirpc/http/capture_client.go

```diff
@@ -56,44 +62,102 @@ type CapturedRequest struct {
 	// RequestURIBytes is the request uri bytes. Notably, this will not include
 	// RequestURI's set by SetRequestURI
 	RequestURIBytes []byte
+	// Metrics is the metrics handler
+	Handler metrics.Handler
 }
 
 var _ Client = &CaptureClient{}
 
 // SetBody stores the body for testing.
 func (c *CapturedRequest) SetBody(body []byte) Request {
+	_, span := c.Handler.Tracer().Start(
```

Comment:
i wouldn't do this here. I'd have a single trace around when `Do()` si called and set the attributes there (`http.body`) specifically

Reply:
fixed